From f83f0a96b2a31f1b3700755fec7ba06753e7be5b Mon Sep 17 00:00:00 2001
From: Tommy Lin <lin.tommysc@inventec.com>
Date: Mon, 5 Dec 2022 07:19:45 +0000
Subject: [PATCH 17/27] Fix excerpt query fail

Symptom/Reason:
- The excerpt query fail

Root Cause:
- Design change in bmcweb caused inventec query parameter patch can no longer apply

Solution/Change:
- [bmcweb]
  - Add hanlder for excerpt keyword

Entry Test:
- curl -X GET https://${bmc}/redfish/v1/Managers/bmc/EthernetInterfaces/eth0/ -G -d '$excerpt'

---
 redfish-core/include/utils/query_param.hpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/redfish-core/include/utils/query_param.hpp b/redfish-core/include/utils/query_param.hpp
index 96885ebb..70ed0992 100644
--- a/redfish-core/include/utils/query_param.hpp
+++ b/redfish-core/include/utils/query_param.hpp
@@ -436,6 +436,10 @@ inline std::optional<Query> parseParameters(boost::urls::params_view urlParams,
                 return std::nullopt;
             }
         }
+        else if (it.key == "$excerpt")
+        {
+            // do nothing
+        }
         else
         {
             // Intentionally ignore other errors Redfish spec, 7.3.1
-- 
2.34.1

